src/libraries/System.Net.Http/src/System/Net/Http/HttpRequestOptions.cs

```diff
@@ -9,9 +10,13 @@ namespace System.Net.Http
     /// <summary>
     /// Represents a collection of options for an HTTP request.
     /// </summary>
-    public sealed class HttpRequestOptions : IDictionary<string, object?>
+    public sealed class HttpRequestOptions : IDictionary<string, object?>, IReadOnlyDictionary<string, object?>
```

Comment:
You'll need to update the ref assembly as well:
https://github.com/dotnet/runtime/blob/3c67669e727a28fe164b50c1df76f4fe6e9d4c7b/src/libraries/System.Net.Http/ref/System.Net.Http.cs